docs/rfc/0698-cli-infrastructure.adoc

```diff
@@ -0,0 +1,298 @@
+
+<<arch-sigs, RFC-682#Signatures>> outlined the signing of Radicle data. This is essential to
+this RFC as commands that write to the storage generally need access
+to the secret key, whether for signing data or proof of ownership to
+the storage. For most (if not all) read-only commands, no access to
+the secret key is needed.
+
+To achieve the goals of having a read-only storage and using
+`<<ssh-agent>>` for signing, we propose the following should be
+implemented:
+
+. `ssh-agent` compatability with `<<radicle-keystore>>`, which has been
+started in `<<rk-17, radicle-keystore#17>>`.
+. A subcommand `rad-profile-ssh-add` for adding one's secret key to the
+`ssh-agent` using the supplied constraints, e.g. ask for password.
+
+== Library/Command Organisation
+
+As proposed here, `radicle-link` will provide a core set of
+functionality. Something this document wishes to standardise is the
+organisation of these components. The author sees two options:
+
+. One binary that defines the core subcommands
+. Many binaries that take advantage of the subcommand mechanism
+
+Pros of *1.*:
+
+* One-stop shop for all core functionality
+* One binary is packaged and distributed
+
+Cons of *1.*:
+
+* Making changes to one subset of the commands could mean a version
+  change of the whole binary
+
+Pros of *2.*:
+
+* We will be organising commands as libraries (as per RFC-682) — so
+  modular commands would mirror this structure well.
+* Subcommands can be versioned independently of each other.
+* We can dogfood our subcommand functionality
+
+Cons of *2.*:
+
+* Makes it easier to override one of the core commands.
+* Many binaries must be packaged and distributed.
+
+The author recommends *1.*. As mentioned, we have a hard requirement
+of not allowing the core commands to be overridden. In *2.* this is
+_always_ possible.
+
+=== `radicle-bins`
+
+Per <<arch-cli, RFC-682#CLI>>, all subcommands will be provided as
+libraries. However, this leaves the question of where the binary might
+live. So far, it has been an unspoken formalism that all binaries live
+in `radicle-bins` where we can provide a `Cargo.lock` file. All
+functionality will live in `radicle-link`, including the `main`
+function, and the entry in `radicle-bins` will be a thin wrapper
+around this function.
+
+== Core
+
+[quote]
+------
+We swears to serve the master of the precious. We will swear on
+the…on the precious.
+------
+-- Gollum, The Two Towers
+
+With all the details in place, we define a non-exhaustive list of the
+core commands that we wish to include for the first iteration. More
+components MAY be added as the result of future RFCs. We SHALL keep a
+list of the core commands as part of the `radicle-link` documentation.
+
+=== `rad-identities`
+
+The `rad identities` subcommand provides the management of identity
+documents and their metadata. It is further split into subcommands
+which we sketch out below:
+
+* `project [ create | get | list | update | checkout | review ]` --
```

Comment:
Note to self: the include file management is needed as well, for managing remotes.

Reply:
Actually, I'm just outright missing `track` and `untrack` :facepalm:

Reply:
Nevermind, I _do_ have tracking which will have the side-effect of adding or removing from the include files